src/promptflow/promptflow/batch/_batch_engine.py

```diff
@@ -254,3 +269,13 @@ def _persist_outputs(self, outputs: List[Mapping[str, Any]], output_dir: Path):
         """Persist outputs to json line file in output directory"""
         output_file = output_dir / OUTPUT_FILE_NAME
         dump_list_to_jsonl(output_file, outputs)
+
+    def _has_running_loop(self) -> bool:
+        """Check if the current thread has a running event loop."""
+        # When using asyncio.get_running_loop(), a RuntimeError is raised if there is no running event loop.
+        # So we use a try-catch block to determine whether there is currently an event loop in place.
```

Comment:
Note that this is the only way to check whether there is a running loop now, see https://docs.python.org/3/library/asyncio-eventloop.html?highlight=get_running_loop#asyncio.get_running_loop

Reply:
Updated.